api/src/main/java/org/regenstrief/linkage/io/OpenMRSReader.java

```diff
@@ -62,26 +64,29 @@
     
     private Collection<String> projections;
 
-    /**
-     * 
-     */
+    private Date reportCreatedOn;
+
     public OpenMRSReader() {
-    	this(null);
+    	this(null,null);
     }
     
-    public OpenMRSReader(Collection<String> projections) {
+    public OpenMRSReader(Collection<String> projections, Date reportCreatedOn) {
         pageNumber = 0;
         resultMode = MODE_PATIENT;
         expand_patient = true;
         if (projections != null && !projections.contains("org.openmrs.Patient.patientId")) {
         	projections.add("org.openmrs.Patient.patientId");
         }
-        this.projections = projections;
-        
-    	log.info("Getting all patient records ...");
-    	updatePatientList();
-    	
-    	log.info("Finish intialization ...");
+        this.projections = null;
+        this.reportCreatedOn = reportCreatedOn;
+    }
+
+    public void init(){
+        log.info("Getting all patient records ...");
+
+        updatePatientList();
+
+        log.info("Finish intialization ...");
```

Comment:
Assuming this is noting completion of initialization, perhaps "Finished initialization."